### ISSUE TITLE
Ideally `show()` should return an `Iterator`

### ISSUE THREAD

**Issue by brainplot** [EstebanBorai/network-interface#45]
Returning some iterator type instead of the whole `Vec` directly would allow consumers to have more control over how memory is allocated.

**Comment by EstebanBorai** [comment]
Hi @brainplot! Thanks for your interest in this crate!

I think returning an iterator is more flexible yet performant.

Perhaps we could have a major release using that approach instead of allocating the whole vector of network interfaces.

Are you interested in working on this?

**Comment by brainplot** [comment]
> Are you interested in working on this?

Yes, I can try to have a crack at this. I'm considering this crate for a project I'm working on.
